samza-core/src/main/java/org/apache/samza/runtime/LocalContainerRunner.java

```diff
@@ -98,8 +102,16 @@ public void onContainerFailed(Throwable t) {
             }
           });
 
+      containerHeartbeatMonitor = getContainerHeartbeatMonitor();
```

Comment:
Can re-name all `getXXX()` to `createXXX()` if all they are doing is creating a new object. OTOH, we should retain `getXX()` if we're returning a cached copy